core/src/main/java/org/elasticsearch/common/network/InetAddresses.java

```diff
@@ -354,4 +359,59 @@ private static InetAddress bytesToInetAddress(byte[] addr) {
             throw new AssertionError(e);
         }
     }
+
+    /**
+     * Computes the min &amp; max ip addresses (represented as long values - same way as stored in index) represented by the given CIDR mask
+     * expression. The returned array has the length of 2, where the first entry represents the {@code min} address and the second the {@code max}.
+     * A {@code -1} value for either the {@code min} or the {@code max}, represents an unbounded end. In other words:
+     *
+     * <p>
+     * {@code min == -1 == "0.0.0.0" }
+     * </p>
+     *
+     * and
+     *
+     * <p>
+     * {@code max == -1 == "255.255.255.255" }
+     * </p>
+     */
+    public static long[] cidrMaskToMinMax(String cidr) {
+        Matcher matcher = MASK_PATTERN.matcher(cidr);
+        if (!matcher.matches()) {
+            return null;
+        }
+        int addr = (( Integer.parseInt(matcher.group(1)) << 24 ) & 0xFF000000)
```

Comment:
I don't think that we should move this code into the InetAddresses.java source file. That code is from the Guava code base, is licensed to the Guava authors (see the license header). By moving this code which is not from Guava here we will create a confusing situation with respect to the licensing of the code.